src/main/java/io/stargate/sgv2/jsonapi/service/operation/reranking/FeatureUsage.java

```diff
@@ -0,0 +1,125 @@
+package io.stargate.sgv2.jsonapi.service.operation.reranking;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a collection of {@link Feature}s in a command. This class is immutable and can be used
+ * to track which features are utilized in a command. This class is immutable; methods that modify
+ * the set of features return a new {@code FeatureUsage} instance. It uses an {@link EnumSet}
+ * internally for efficient storage and operations on features.
+ */
+public final class FeatureUsage {
```

Comment:
I think we can change this from Immutable (which is normally a good idea) to mutable given the way it is used in the CommandContext. We are collecting the feature usage as we go, so it is designed to be mutated. 

Also, because this class is wrapping the EnumSet we are controlling the mutations. i.e. it is not possible to accidently remove or clear the set. 

Often we choose Immutable structures because (IMHO) most bugs come from unexpected state mutation. This class is designed to control the state mutation.